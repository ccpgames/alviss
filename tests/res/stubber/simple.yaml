foo: str
internal_ref:
  deeper:
    there: str
    everywhere: str*
base_key: INT
foo_inherited: float
bar: float*
group:
  alpha: Str | iNt
  beta: float|int|str*
  gamma_with_var: str|none
  delta: int|None
collapsed:
  other_group: dict|None
  group:
    successful: bool
    in:
      one:
        string: int
my_list: list[str]
import_test:
  imported_keys: Dict[str,str]
should_be_seven:
  sub_seven: dict*

