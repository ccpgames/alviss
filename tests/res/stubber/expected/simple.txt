__all__ = [
    'CfgInternalRefDeeperStub',
    'CfgInternalRefStub',
    'CfgGroupStub',
    'CfgCollapsedGroupInOneStub',
    'CfgCollapsedGroupInStub',
    'CfgCollapsedGroupStub',
    'CfgCollapsedStub',
    'CfgImportTestStub',
    'CfgShouldBeSevenStub',
    'CfgStub',
    'AlvissConfigStub',
]

from typing import *
from alviss.structs import Empty
from alviss.structs.cfgstub import _BaseCfgStub
from alviss.structs import BaseConfig


class CfgInternalRefDeeperStub(_BaseCfgStub):
    there: Union[str, Empty]
    everywhere: str


class CfgInternalRefStub(_BaseCfgStub):
    deeper: CfgInternalRefDeeperStub


class CfgGroupStub(_BaseCfgStub):
    alpha: Union[str, int, Empty]
    beta: Union[float, int, str]
    gamma_with_var: Union[str, None, Empty]
    delta: Union[int, None, Empty]


class CfgCollapsedGroupInOneStub(_BaseCfgStub):
    string: Union[int, Empty]


class CfgCollapsedGroupInStub(_BaseCfgStub):
    one: Union[CfgCollapsedGroupInOneStub, Empty]


class CfgCollapsedGroupStub(_BaseCfgStub):
    successful: Union[bool, Empty]
    in: Union[CfgCollapsedGroupInStub, Empty]


class CfgCollapsedStub(_BaseCfgStub):
    other_group: Union[Dict[str, Any], None, Empty]
    group: Union[CfgCollapsedGroupStub, Empty]


class CfgImportTestStub(_BaseCfgStub):
    imported_keys: Union[Dict[str, str], Empty]


class CfgShouldBeSevenStub(_BaseCfgStub):
    sub_seven: Dict[str, Any]


class CfgStub(_BaseCfgStub):
    foo: Union[str, Empty]
    internal_ref: CfgInternalRefStub
    base_key: Union[int, Empty]
    foo_inherited: Union[float, Empty]
    bar: float
    group: CfgGroupStub
    collapsed: Union[CfgCollapsedStub, Empty]
    my_list: Union[List[str], Empty]
    import_test: Union[CfgImportTestStub, Empty]
    should_be_seven: CfgShouldBeSevenStub


class AlvissConfigStub(BaseConfig, CfgStub):
    pass
